esmvalcore/preprocessor/_supplementary_vars.py

```diff
@@ -101,39 +101,72 @@ def add_cell_measure(
     )
 
 
-def add_ancillary_variable(cube: Cube, ancillary_cube: Cube) -> None:
+def add_ancillary_variable(
+    cube: Cube, ancillary_cube: Cube | iris.coords.AncillaryVariable
+) -> None:
     """Add ancillary variable to cube (in-place).
 
-    Note
-    ----
-    This assumes that the ancillary variable spans the rightmost dimensions of
-    the cube.
-
     Parameters
     ----------
     cube:
         Iris cube with input data.
     ancillary_cube:
-        Iris cube with ancillary data.
+        Iris cube or AncillaryVariable with ancillary data.
 
     Returns
     -------
     iris.cube.Cube
         Cube with added ancillary variables
     """
-    coord_dims = tuple(range(cube.ndim - len(ancillary_cube.shape), cube.ndim))
-    ancillary_data = ancillary_cube.core_data()
+    try:
+        ancillary_var = iris.coords.AncillaryVariable(
+            ancillary_cube.core_data(),
+            standard_name=ancillary_cube.standard_name,
+            units=ancillary_cube.units,
+            var_name=ancillary_cube.var_name,
+            attributes=ancillary_cube.attributes,
+        )
+    except AttributeError as err:
+        msg = (
+            f"Failed to add {ancillary_cube} to {cube} as ancillary var."
+            "ancillary_cube should be either an iris.cube.Cube or an "
+            "iris.coords.AncillaryVariable object."
+        )
+        raise ValueError(msg) from err
+    data_dims: list[int | None] = []
+    if isinstance(ancillary_cube, iris.coords.AncillaryVariable):
+        start_dim = cube.ndim - len(ancillary_var.shape)
+        data_dims = list(range(start_dim, cube.ndim))
+    else:
```

Comment:
Could you add a little comment here what this code does? E.g., match coordinates of ancillary cube to dimensions of data cube to determine dimensions of ancillary data.